radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixJsonRpcClient.java

```diff
@@ -77,19 +73,19 @@ public RadixJsonRpcClient(WebSocketClient wsClient) {
 			.publish()
 			.refCount();
 
-		if (!CHECK_API_VERSION) {
-			this.serverApiVersion = Single.just(API_VERSION);
-		} else {
-			this.serverApiVersion = jsonRpcCall("Api.getVersion")
-				.map(result -> result.getAsJsonObject().get("version").getAsInt())
-				.cache();
-		}
+		this.serverApiVersion = SingleSubject.create();
+		this.serverApiVersion.onSuccess(API_VERSION);
+		jsonRpcCall("Api.getVersion")
+			.map(result -> result.getAsJsonObject().get("version").getAsInt())
+			.subscribe(serverApiVersion::onSuccess, e -> LOGGER.error("Error while requesting Api.getVersion: " + e, e));
```

Comment:
This looks wrong to me. `onSuccess` should only ever be called once and is called twice here no?

Reply:
As discussed, this was a hack as the core currently doesn't support Api.getVersion. I've improved the hack until the core fixes it to return a compatible API version on error.